Develop

eusatenko opened the pull request:
Готово, готов к критике.

eusatenko committed:
Добавил класс для тестов на создание курьера.
Тесты: создание, создание 2-х одинаковых.
Добавил удаление созданного курьера в @After

eusatenko committed:
Готово, добавил классы и тесты на проверку создания нового Курьера, авторизации Курьером, создание заказа, получения списка заказов.

eusatenko committed:
add allure report

alesyaisachenko commented on src/test/java/Courier.java:
```diff
@@ -0,0 +1,73 @@
+public class Courier {
+//    private File json;
+
+//    метод создания курьера и получения тела ответа
```
используй аннотацию [@Step](https://github.com/step), для того чтобы отчеты в алюр были более информативные.

eusatenko replied:
OK

alesyaisachenko commented on src/test/java/Courier.java:
```diff
@@ -0,0 +1,73 @@
+                .and()
+                .body(json)
+                .when()
+                .post("/api/v1/courier/");
```
"/api/v1/courier" -- лучше было вынести в final static переменную.

eusatenko replied:
OK

alesyaisachenko commented on src/test/java/Courier.java:
```diff
@@ -0,0 +1,73 @@
+                .header("Content-type", "application/json")
+                .auth().oauth2("")
+                .and()
+                .body(json)
```
в боди передавать данные нужно через сериализацию. Это более продвинутый вариант, в теории он описан.

Это нужно обязательно пофиксить во всем коде.

eusatenko replied:
OK

alesyaisachenko commented on src/test/java/Courier.java:
```diff
@@ -0,0 +1,73 @@
+//    проверка создания курьера
+    public void CheckCreateNewCourier(File json) {
+        // создаём курьера и записываем ответ
+        Response response =
```
это у тебя реализовано в методе createNewCourierAndGetResponse(File json) .

Дублирование кода нужно удалить.

eusatenko replied:
OK

alesyaisachenko commented on src/test/java/Courier.java:
```diff
@@ -0,0 +1,73 @@
+        // проверяем ответ
+        response.then().assertThat().body("ok",equalTo(true))
+                .and()
+                .statusCode(201);
```
статус коды можно было взять из библиотеки:
import  static  org.apache.http.HttpStatus.*;

statusCode(SC_CREATED)

eusatenko replied:
OK
Правда не очень понятно зачем это надо?
С кодами проще, но это мне -)

alesyaisachenko commented on src/test/java/Courier.java:
```diff
@@ -0,0 +1,73 @@
+
+    public int getIdCourier(File json) {
+        int idCourier = given().
+                contentType("application/json").
```
часть кода реализовано в методе getResponseLoginCourier(File json).

Дублирование кода нужно удалить

eusatenko replied:
ОК

alesyaisachenko commented on src/test/java/ListOfOrdersTest.java:
```diff
@@ -0,0 +1,27 @@
+@DisplayName("Получение списка заказов")
+public class ListOfOrdersTest {
+    @Before
+    public void setUp() {
```
блок @Before у тебя дублируются в каждом тесте.
 
Дублирование кода нужно избегать.
 
Как это можно сделать:
 
Создать класс BaseTest, в него вынести блоки @before и @after. Остальные тестовые классы наследовать от BaseTest.
 
Это нужно обязательно пофиксить.

eusatenko replied:
ОК частично.
Вынес блок @Before и наследую класс.
@After вынести не смог, так как 1 - передаю в метод переменную из теста, 2 - не во всех тестовых классах он есть.

alesyaisachenko commented on src/test/java/ListOfOrdersTest.java:
```diff
@@ -0,0 +1,27 @@
+    @DisplayName("Тест получения списка заказов")
+    @Description("Получаем список заказов")
+    public void checkListOfOrderNotNull() {
+        given()
```
Если следовать твоей архитектуре кода, то для формирования запросов для Order нужно также создать отдельный класс и разместить там методы для формирования запросов (как ты сделал с классом Courier).

Это нужно обязательно пофиксить.

eusatenko replied:
ОК

alesyaisachenko commented on src/test/java/LoginCourierTest.java:
```diff
@@ -0,0 +1,106 @@
+    @DisplayName("Тест авторизации Курьера")
+    @Story("Логин курьера")
+    @Description("курьер может авторизоваться, успешный запрос возвращает id")
+    public void CheckLoginCourier() {
```
CheckLoginCourier()  -- названия методов должны начинаться только с маленькой буквы.

Это нужно обязательно пофиксить во всем коде.

eusatenko replied:
ОК

alesyaisachenko commented on src/test/java/OrderTest.java:
```diff
@@ -0,0 +1,69 @@
+                "2025-06-06",
+                "как можно быстрее",
+                new String[]{color.toString()});
+        Response response =
```
код по формированию запроса нужно вынести в отдельный класс.

eusatenko replied:
ОК

eusatenko committed:
1) Разнёс классы по разным пакетам.
2) Устранил замечания, но не понял до конца как сделать такой момент:
Вынес блок @before и наследую класс - в разных классах использую его через @Ovverade (если надо изменить).
@after вынести не смог, так как 1 - передаю в метод переменную из теста, 2 - не во всех тестовых классах он нужен.

eusatenko commented:
Подскажите, пожалуйста.
Можно ли как-то заставить аллюр написать данные в читабельном виде в параметре метода теста?
Или это только если передавать простые типы переменных (строка, число и т.д.)?

В отчёте Allure, в тесте, в теле теста (Test body) пишется метод - раскрыть его - параметр и его значение.
Пример: 
Создание курьера и получение ответа 1 parameter108ms
courierData	data.CourierData@32a074ed
<img width="1201" alt="image" src="https://user-images.githubusercontent.com/84189497/207709952-b6c32fca-3041-406e-b1b3-7abc9c298e7b.png">

eusatenko commented:
Проект принят.
